cmd/cstor-volume-mgmt/controller/volume-controller/run_volume_controller.go

```diff
@@ -0,0 +1,102 @@
+package volumecontroller
+
+import (
+	"fmt"
+	"time"
+
+	"github.com/golang/glog"
+	"github.com/openebs/maya/cmd/cstor-volume-mgmt/controller/common"
+	"k8s.io/apimachinery/pkg/util/runtime"
+	"k8s.io/apimachinery/pkg/util/wait"
+	"k8s.io/client-go/tools/cache"
+)
+
+// Run will set up the event handlers for types we are interested in, as well
+// as syncing informer caches and starting workers. It will block until stopCh
+// is closed, at which point it will shutdown the workqueue and wait for
+// workers to finish processing their current work items.
+func (c *CStorVolumeController) Run(threadiness int, stopCh <-chan struct{}) error {
+	defer runtime.HandleCrash()
+	defer c.workqueue.ShutDown()
+
+	// Start the informer factories to begin populating the informer caches
+	glog.Info("Starting CStorVolume controller")
+
+	// Wait for the k8s caches to be synced before starting workers
+	glog.Info("Waiting for informer caches to sync")
+	if ok := cache.WaitForCacheSync(stopCh, c.cStorVolumeSynced); !ok {
+		return fmt.Errorf("failed to wait for caches to sync")
+	}
+	glog.Info("Starting CStorVolume workers")
+	// Launch worker to process CStorVolume resources
+	for i := 0; i < threadiness; i++ {
```

Comment:
Do we need to manage the cases where this controller will be running as multiple instances i.e. more than one replicas when deployed as a kubernetes Deployment.

Reply:
Not sure what we need to do in that case. Any suggestion ?